.github/workflows/build.yml

```diff
@@ -98,100 +98,77 @@ jobs:
         env:
           GITHUB_EVENT: ${{ toJson(github.event) }}
 
-  build:
-    runs-on: ${{ matrix.config.build_machine }}
-    if: ${{ needs.generate-matrices.outputs.packages-with-dependents != '[]' }}
-    name: ${{ matrix.config.host_profile }} ${{ matrix.package.package_reference }}
+  build_0:
```

Comment:
I wonder if this could be deduplicated by running one job multiple times? but I guess this is something github actions does not support maybe?

or a build matrix with a counter and dependent points to the previous one?

Reply:
Jobs in build_1 are depending on jobs in build_0. If we move all to just build, without _0 and _1, it's impossible to do dependency link, because "needs" keyword does not understand matrices. Maybe someday github will allow adding jobs in a for loop, but I don't have high hopes for such functionality.